trappy/base.py

```diff
@@ -200,13 +202,20 @@ def generate_parsed_data(self):
             # When running out of memory, Pandas has been observed to segfault
             # rather than throwing a proper Python error.
             # Look at how much memory our process is using and warn if we seem
-            # to be getting close to the system's limit.
+            # to be getting close to the system's limit, check it only once
+            # in the beginning and then every CHECK_MEM_COUNT events
+            check_memory_count -= 1
+            if check_memory_count != 0:
```

Comment:
OK, I tested it and pandas still segfaults when you run out of memory, so we'll need to keep this hack.